### ISSUE TITLE
master-pd is not correctly deleted after the test ends

### ISSUE THREAD

**Issue by gmarek** [kubernetes/kubernetes#32366]
I saw this happen twice: once in "kubemark-100-high-density" suite yesterday, and now in a PR-builder.
The effect is following:
```
Starting master and configuring firewalls
ERROR: (gcloud.compute.disks.create) Some requests did not succeed:
 - The resource 'projects/k8s-jkns-pr-gce/zones/us-central1-f/disks/e2e-gce-agent-pr-50-0-master-pd' already exists

2016/09/09 02:34:43 e2e.go:455: Error running up: exit status 1
```

cc @kubernetes/test-infra-maintainers, assigning @fejta to route to the correct person.

**Comment by jsafrane** [comment]
another hit: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/31607/kubernetes-pull-build-test-e2e-gce/58017/

**Comment by wojtek-t** [comment]
And one more:
http://kubekins.dls.corp.google.com:8080/view/Scalability/job/kubernetes-e2e-gce-enormous-cluster/146/console

**Comment by gmarek** [comment]
I'll delete dangling disks to deflake pr builder.

**Comment by gmarek** [comment]
Actually it deleted itself eventually...

**Comment by bprashanth** [comment]
I see this from tiem to time as well, usually there's one run where: 
```
17:40:26 ERROR: (gcloud.compute.copy-files) Could not fetch instance:
17:40:26  - The resource 'projects/kubernetes-ingress/zones/us-central1-f/instances/jenkins-e2e-master' was not found
17:40:26 
```

Which is a bogus error because 2 lines before we see
```
17:39:49 NAME                   ZONE           SIZE_GB  TYPE    STATUS
17:39:49 jenkins-e2e-master-pd  us-central1-f  20       pd-ssd  READY
17:39:49 
```
And subsequent tests fail with: 
```
05:00:40 ERROR: (gcloud.compute.disks.create) Some requests did not succeed:
05:00:40  - The resource 'projects/kubernetes-ingress/zones/us-central1-f/disks/jenkins-e2e-master-pd' already exists
05:00:40 
05:00:40 2016/09/09 05:00:40 e2e.go:455: Erro
```

The larger issues is actually similar to b/27367941, but maybe we should just retry harder?

**Comment by bprashanth** [comment]
Oh, i missed something, we actually created the pd, then tried to allocate static ip which gave a 500, so leaked the pd

logs from the first failure: http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce-ingress/11101/console
```
17:39:53 Created [https://www.googleapis.com/compute/v1/projects/kubernetes-ingress/regions/us-central1/addresses/jenkins-e2e-master-ip].
17:39:55 ERROR: (gcloud.compute.addresses.describe) Could not fetch resource:
17:39:55  - Internal Error
17:39:55 
```

**Comment by quinton-hoole** [comment]
This broke the multizone (federation-lite) tests too:

http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce-multizone/3158

I deleted the dangling disk, and that unbroke the test, at least temporarily.

**Comment by quinton-hoole** [comment]
I've added a v1.4 release blocker label to this, as I don't think that we can release without fixing it and demonstrating that e2e tests are passing consistently.

cc: @pwittrock FYI.

**Comment by ixdy** [comment]
It looks like a bug in the teardown step. It should be deleting the preexisting disk.

**Comment by ixdy** [comment]
I think I found the bug.

We look for disks machine the replica name during teardown (https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/util.sh#L1185-L1192), but we get the replica name by looking at running instances (https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/util.sh#L1321-L1327). So if the pd has been stranded, we won't delete it.

**Comment by ixdy** [comment]
Bug introduced in #29995. cc @jszczepkowski @zmerlynn @wojtek-t

**Comment by quinton-hoole** [comment]
Thanks @ixdy!  Awesome debugging!

**Comment by gmarek** [comment]
It seems that it wasn't fixed. cc @jszczepkowski @ixdy

**Comment by quinton-hoole** [comment]
@gmarek This is fixed in head.  See recent PR by @colhom

**Comment by pwittrock** [comment]
Why is this a release blocker?

**Comment by gmarek** [comment]
Our startup scripts are wrong. But I think it's fixed yesterday. @quinton-hoole ?

**Comment by madhusudancs** [comment]
@gmarek not yet. This is the PR - https://github.com/kubernetes/kubernetes/pull/33039.

**Comment by pwittrock** [comment]
As no one has given me a justification for this to be required in release 1.4 and we are no longer accepting additional changes to it, I am moving this out of the milestone.

**Comment by pwittrock** [comment]
@madhusudancs @jszczepkowski @gmarek

**Comment by madhusudancs** [comment]
@pwittrock Ack. As pointed out here - https://github.com/kubernetes/kubernetes/pull/33039#issuecomment-248741672 this bug shouldn't be happening in v1.4. So there is nothing to be fixed in v1.4 for this issue.

**Comment by gmarek** [comment]
Fixed by #33039
